ssawg_trending_scraper.py

```diff
@@ -0,0 +1,376 @@
+# --------------------------------------------------
+# Scrape the SSAWG/Aspect pages for trending data
+# and copy them to a separate page for consolidation
+# and efficient viewing.
+# ---------------------------------------------------
+
+import requests
+import netrc
+import argparse
+import Ska.ftp
+from bs4 import BeautifulSoup
+from urllib import request, response, error, parse
+from urllib.request import urlopen
+from datetime import datetime, timedelta
+import Chandra.Time
+import re
+
+
+# ------------------------------------
+# Setup for password-protected site(s)
+# ------------------------------------
+
+netrc = Ska.ftp.parse_netrc()
+if 'https://cxc.cfa.harvard.edu' not in netrc:
+    raise RuntimeError('must have authentication in .netrc')
+
+user = netrc['https://cxc.cfa.harvard.edu']['login']
+password = netrc['https://cxc.cfa.harvard.edu']['password']
+
+# -----------------------------------
+# ESTABLISH BASE URL, TRENDING PAGES
+# AND CREATE PROPER URLS TO USE
+# -----------------------------------
+
+base_url = "https://cxc.cfa.harvard.edu/mta/ASPECT/"
+
+trending_pages = [
+    "kalman_watch/", 
+    "acq_stat_reports/",
+    "gui_stat_reports/",
+    "perigee_health_plots/",
+    "periscope_drift_reports/", # this page requires a username/password
+    "obc_rate_noise/trending/",
+    "fid_drift/",
+    "aimpoint_mon/",
+    "celmon/",
+    "vv_rms/",
+    "attitude_error_mon/",
+    "fss_check3/"
+]
+
+
+def proper_url(page, base_url):
```

Comment:
Yeah, I think I started to figure that out later on but wrote this one earlier. It shall be corrected.

Reply:
I'm avoiding creating subclasses at this point, just because I've restructured the code significantly. But all other comments/suggestions were implemented.